Lesson2

VeronikaMads opened the pull request:
Основное задание по уроку 2

VeronikaMarss committed:
коммит второго задания

VeronikaMarss committed:
коммит второго задания - файл

pavelTMS commented on Lesson2/src/Task1.java:
```diff
@@ -0,0 +1,14 @@
+    public static void main(String[] args) {
+        int count = 0;
+        for (String arg : args) {
+            System.out.println();
```
System.out.println(); это можно удалить. лишнее действие

pavelTMS commented on Lesson2/src/Task1.java:
```diff
@@ -0,0 +1,14 @@
+        int count = 0;
+        for (String arg : args) {
+            System.out.println();
+            int i = Integer.parseInt (arg);
```
`Integer.parseInt (arg);` между parseInt и ( пробел. Идея должна ругаться и код не запустит

pavelTMS commented on Lesson2/src/Task1.java:
```diff
@@ -0,0 +1,14 @@
+        }
+        System.out.println(count);
+    }
+}
```
вы также засушили Task1.class они в гите не нужны.
что бы они не добавлялись надо добавить в файл .gitignore (.*class) и еще папку .idea

VeronikaMarss committed:
ДЗ по второму уроку.
1. Проверка у меня не запускается, не понимаю почему.. смотрела по лекции.
2. Второе задание скорее всего тоже не верно, но может вы подталкнете куда двигатся.
3. Не могу понять как в одном классе работают несколько ДЗ (задач), В одном классе мы создаем методы и в них выполняются задачи ?
4. Не смогла примерить функцию debug. Если я выделаю строку и наживаю - происходит тоже самое что и нажимая run..

р.s буду еще раз пересматривать лекции 2 и 3.

pavelTMS commented on Lesson2/src/homework2.java:
```diff
@@ -0,0 +1,33 @@
+        int a = 100;
+        int b = 200;
+        int c = a + b;
+        System.out.println(c);
```
у вас есть класс homework2 - классы принято называть что бы первая буква была большой HomeWork2.
далее вы объявили переменые а и б и присвоили им значения 100 и 200 далее в переменную с записали сумму а и б.
далее вывели на консоль значение переменной с.
все это мы делали в методе main - это входная точка в программу.
в методе main мы можем вызвать другие методы.
например:
у вас есть ниже метод c(int a, int b) - который принимает на вход две переменные а и б примитивного типа инт
что бы вызвать метод c(int a, int b) из метода main
надо в строчке 9 написать с(100, 200);

pavelTMS commented on Lesson2/src/homework2.java:
```diff
@@ -0,0 +1,33 @@
+        return (int) c;
+    }
+
+    public static void main(String[] args) {
```
два метода main не может быть в одном классе. надо создать новый класс HomeWork3 и туда перенести этот метод.

pavelTMS commented on Lesson2/src/homework2.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    public static int c(int a, int b) {
+        long c = a + (Long) b;
```
long c = a + (Long) b; тут ошибка
<img width="202" alt="image" src="https://user-images.githubusercontent.com/86020312/123483150-f8fabc00-d60e-11eb-9906-3d6a15ae4344.png">
есть примитивный тип long - маленькими буквами написан и есть 
обертка Long - сылочный тип данных(это уже класс целый в него можно зайти контрол+ клик мышкой). так вот вы приводите примитивную переменую к сылочному типу Long - поэтому идия и ругается
надо просто приводить к примитивному типу long(который маленькими буквами пишется)
long c = a + (long) b;

pavelTMS commented on Lesson2/src/homework2.java:
```diff
@@ -0,0 +1,33 @@
+import static java.lang.Math.max;
```
4)что бы запустить дебаг нужно поставить точку остановки например в методе main - это точка коричневого цвета. мышкой кликаете она появляется
<img width="435" alt="image" src="https://user-images.githubusercontent.com/86020312/123483730-e2a13000-d60f-11eb-8152-80c97e14c148.png">
затем правой клавишей мыши на классе
<img width="772" alt="image" src="https://user-images.githubusercontent.com/86020312/123483838-0f554780-d610-11eb-98b1-dff96b63c1c0.png">

pavelTMS commented on Lesson2/src/homework2.java:
```diff
@@ -0,0 +1,33 @@
+import static java.lang.Math.max;
+
```
3) есть класс Main
в классе может быть много методов method1, method2, method3
<img width="442" alt="image" src="https://user-images.githubusercontent.com/86020312/123484134-8c80bc80-d610-11eb-8654-50f93bd4d66e.png">

есть точка входа наш любимый метод main в нем можно и вызвать все эти методы.
<img width="446" alt="image" src="https://user-images.githubusercontent.com/86020312/123484346-dc5f8380-d610-11eb-8978-752dc269369c.png">

pavelTMS replied:
и на консоль выведутся строки
"вызов метода 1"
"вызов метода 2"
"вызов метода 3"

pavelTMS commented on Lesson2/src/homework2.java:
```diff
@@ -0,0 +1,33 @@
+import static java.lang.Math.max;
+
+public class homework2 {
```
1)Проверка у меня не запускается, не понимаю почему.. --- что за проверка не запускается? debug?

VeronikaMarss committed:
ДЗ по второму уроку /2
1. в первой задаче исправила ошибки.
2. в второй задаче не смогла написать код с помощью boolean. (подскажите в чем ошибка) и проверку пока не понимаю как сдать.
3. четвертая задача решина с помощью интернета.
4. за 5 еще не садилась. попываюсь сегодня/завтра.

VeronikaMarss committed:
ДЗ по второму уроку /3
1. Задачу 2 исправила. Так же проработала с "if" спасибо!.
2. В задаче 3 исправила. Замелила цифру "5" на "длину массива". Код отформатировала через "reformat code"
3. В задаче 4 так же довала скобки в for. и отформатировала через "reformat code".
4. + решение последней задачи № 5.
ДЗ по Уроку 3 сегодня буду разбирать. Сбрасывать !

pavelTMS commented on Lesson2/src/HomeWork4.java:
```diff
@@ -0,0 +1,15 @@
+public class HomeWork4 {
+    public static void main(String[] args) {
+        int[] mass = {2, 3, 4, 5, 6};
+        int count = 0;
```
{2, 3, 4, 5, 7} если будет вот такой массив то посчитает не правильно. надо выбрать другой тип данных

pavelTMS commented on Lesson2/src/HomeWork6.java:
```diff
@@ -0,0 +1,15 @@
+        int a = scanner.nextInt();
+        System.out.print("Введите значение катета b: ");
+        int b = scanner.nextInt();
+        int result = ((a * a) + (b * b));
```
гипотенузу не правильно посчитали. надо еще корень квадратный взять от result

VeronikaMarss committed:
ДЗ по второму уроку /4
1. исправила задачи 3 и 5. В двух задачах изменила тип данных. Проверила. Исправила формулу по гипотенузе.
